### PR TITLE
main/dbus: use xvfb-run to run testsuite

### PR DESCRIPTION
**Pull request by maxice8**
@ncopa
@TBK

### CODE REVIEW

File: main/dbus/APKBUILD [hunk 1]
```diff
@@ -6,13 +6,14 @@ pkgdesc="Freedesktop.org message bus system"
 url="https://www.freedesktop.org/Software/dbus"
 pkggroups="messagebus"
 arch="all"
-options="!check"  # Requires running X11 server.
+# options="!check"  # Requires running X11 server.
```
Comment (tcely) [inline main/dbus/APKBUILD]:
```suggestion
```

**Comment by ncopa** [comment]
looks like dbus fails for some reason

**Comment by Cogitri** [comment]
Because it can't pull in xvfb-run because https://github.com/alpinelinux/abuild/pull/53 isn't merged yet.

**Comment by ncopa** [comment]
I think this will introduce a circular dependency. apparently dbus is needed to be able to build xorg-server.

**Comment by maxice8** [comment]
Disabled on dbus

**Comment by maxice8** [comment]
ping

File: main/dbus/APKBUILD [hunk 2]
```diff
@@ -3,16 +3,17 @@ pkgname=dbus
...
 arch="all"
-options="!check"  # Requires running X11 server.
 license="AFL-2.1 OR GPL-2.0-or-later"
 subpackages="$pkgname-dev $pkgname-doc $pkgname-libs $pkgname-x11 $pkgname-openrc"
 triggers="$pkgname.trigger=/etc/dbus-1/system.d"
 depends_dev="util-linux-dev"
 makedepends="$depends_dev expat-dev libx11-dev autoconf automake libtool xmlto
-	libsm-dev"
+	libsm-dev glib-dev"
+checkdepends="xvfb-run"
```
Comment (tcely) [inline main/dbus/APKBUILD]:
```suggestion
#checkdepends="xvfb-run"
```

Comment (maxice8) [reply main/dbus/APKBUILD]: Check is disabled, checkdepends isn't taken into account.

File: main/libsexy/APKBUILD [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+pkgrel=9
 pkgdesc="Doing naughty things to good widgets."
 url="http://chipx86.com/wiki/Libsexy"
 arch="all"
-options="!check"  # Tests require manual intervention and X11 display.
-license="LGPL-2.0+"
+license="LGPL-2.0-or-later"
 subpackages="$pkgname-dev $pkgname-doc"
 makedepends="gtk+2.0-dev libxml2-dev iso-codes-dev"
+checkdepends="xvfb-run"
```
Comment (tcely) [inline main/libsexy/APKBUILD]: Why was this added if it is not used in `check`?

**Comment by maxice8** [comment]
On Sun, Jun 23, 2019 at 08:18:36AM -0700, tcely wrote:
> tcely commented on this pull request.
> 
> 
> 
> >  subpackages="$pkgname-dev $pkgname-doc"
>  makedepends="gtk+2.0-dev libxml2-dev iso-codes-dev"
> +checkdepends="xvfb-run"
> 
> Why was this added if it is not used in `check`?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/alpinelinux/aports/pull/8426#pullrequestreview-253137451

Fixed

File: main/dbus/APKBUILD (hunk 2 above)
Comment (tcely) [reply main/dbus/APKBUILD]: You're correct. `want_check` prevents this variable from being used.

File: main/libsexy/APKBUILD [hunk 4]
```diff
@@ -28,6 +27,11 @@ build ()
 	make
 }
 
+check() {
+	cd "$builddir"
+	make check
```
Comment (tcely) [inline main/libsexy/APKBUILD]:
```suggestion
	xvfb-run make check
```

File: main/dbus/APKBUILD [hunk 5]
```diff
@@ -3,16 +3,17 @@ pkgname=dbus
...
 pkggroups="messagebus"
 arch="all"
-options="!check"  # Requires running X11 server.
 license="AFL-2.1 OR GPL-2.0-or-later"
 subpackages="$pkgname-dev $pkgname-doc $pkgname-libs $pkgname-x11 $pkgname-openrc"
 triggers="$pkgname.trigger=/etc/dbus-1/system.d"
 depends_dev="util-linux-dev"
 makedepends="$depends_dev expat-dev libx11-dev autoconf automake libtool xmlto
-	libsm-dev"
+	libsm-dev glib-dev"
```
Comment (tcely) [inline main/dbus/APKBUILD]: Is `glib-dev` needed for building tests?
